### ISSUE TITLE
Discover/GHNS: Display Ratings and Reviews, allow to write reviews and rate any item from within KDE Plasma or Discover

### ISSUE THREAD

**Issue by star-buck** [blue-systems/plasma-5.9#193]
currently it looks like this:
![Screenshot_20190514_130458](https://user-images.githubusercontent.com/840070/57693243-092ac980-7649-11e9-995d-eff38f18342e.png)

The numbers before the "Ratings" is the actual Rating Score, not the count of Ratings, so currently one could think there are 63 or 75 Ratings, that is 75 persons rated the app and possibly posted a review comment with it.

This ticket is to gather screenshots and write down what is currently used, and later can make some suggestions how to allow for any item to be rated and reviews from within KDE Plasma or Discover as long as a user is authenticated via kaccounts or else.

**Comment by star-buck** [comment]
in above screenshot it is also not possble to see what Category the search results are from, so if the category is not repeated in the product title, i have no idea if Qogir is a look and feel, Plasma theme or Color scheme...

**Comment by leinir** [comment]
Thought it would be worth mentioning that this was at least partly due to KNS having (as far as i can tell) never correctly reported the number of comments on any item, and instead returning the rating. As of a "few" days ago (that is, Frameworks 5.63), that has changed, and KNS now correctly reports the number of comments (as was originally intended) rather than the rating :)

It seems to have been a simple copy/paste error from when the functionality was first added, and because Discover is the only thing that uses it, and comments were reported weirdly by the server in the past as well, we never quite worked out why it happened, so... Much better now anyway, and we are now in a position where we can more reasonably start working out what we really want to be displaying and whatnot.

...such as, would we want to report the number of ratings there rather than the number of comments, and what sort of effect that might have on users of the api. Arguably we might interpret it like that, but then we would need to make more use of the comments model in KNSCore to get the real number of comments, which would put more load on the server (it would not happen until people showed the entry details, when it sort of already happens anyway, so i'm honestly unsure of whether it would cause any real problems that we don't have already).

All that aside, though, as of Frameworks 5.63, the number showed there is now the actual number of comments as reported by the server, rather than the rating :)

**Comment by leinir** [comment]
Right! Between Nate and i, we've worked out what was going on more precisely. Of course there were multiple levels to the problem ;)

In short:
* Comment entries in the xml returned by the store might have multiple "children" elements, whereas Attica expected exactly one. Fixed in https://phabricator.kde.org/D25888
* Discover's Reviews model (and consequently the view) did not support nested comments. Fixed in https://phabricator.kde.org/D25886

In other words, this will be fixed in 5.66 and 5.17 respectively (both work independently, and combine to make up the fix for this issue).

**Comment by leinir** [comment]
(oops, didn't mean to close it, that's just the presentation side, the "making new comments" bit is not there yet)
